shared_model/backend/plain/impl/peer.cpp

```diff
@@ -28,3 +28,7 @@ const std::optional<shared_model::interface::types::TLSCertificateType>
     &Peer::tlsCertificate() const {
   return tls_certificate_;
 }
+
+void Peer::setTlsCertificate(interface::types::TLSCertificateType cert) {
```

Comment:
Why? Model objects are supposed to be immutable.

Reply:
Because we can't receive both address and TLS from DB in a single call. First enumerate peers with addresses, second check TLS for them. If make it with temporary objects: first get peers and addresses, than TLS and then create peer objects, it will cause to extra memory allocations.